R/user_id.R

```diff
@@ -0,0 +1,78 @@
+#' Mark a user id as a screen name
+#' 
+#' @description 
+#' There are two ways to identify a twitter user: a screen name (e.g.
+#' "justinbieber") or a user identifier (e.g. "27260086"). User identifiers
+#' are 64-bit integers which can't be represented natively by any base R data
+#' structures, so in most cases rtweet returns them as strings. This introduces
+#' an ambiguity, because user names can also consist solely of numbers 
+#' (e.g. "123456"). You can resolve this ambiguity by using `as_screenname()` 
+#' to a character vector a screen name.
+#' 
+#' In general, you are best off using user ids where possible; screen names are
+#' not static and may change. 
+#' 
+#' @param x A character vector of twitter screen names.
+#' @examples
+#' \dontrun{
+#' # Look up user with id 123456 (screen name harperreed)
+#' lookup_users("123456") 
+#' 
+#' # Look up user with name 123456
+#' lookup_users(as_screenname("123456"))
+#' }
+#' @family users
+#' @rdname as_screenname
+#' @export
+as_screenname <- function(x) {
+  stopifnot(is.character(x))
+  structure(x, class = "rtweet_screen_name")
+}
+
+#' @export
+`[.rtweet_screen_name` <- function(x, i) {
+  as_screenname(NextMethod())
+}
+
+#' @export
+print.rtweet_screen_name <- function(x, ...) {
+  cat("<rwteet_screen_name>\n")
+  class(x) <- NULL
+  print(x, ...)
+  invisible(x)
+}
+
+#' @rdname as_screenname
+#' @export
+#' @usage NULL
+as_userid <- function(x) {
+  lifecycle::deprecate_warn("1.0.0", "as_userid()")
+  x
+}
+
+user_type <- function(x, arg_name = "user") {
+  if (is.numeric(x)) {
+    "user_id"
+  } else if (is.character(x)) {
+    if (inherits(x, "rtweet_screen_name")) {
+      # needed for purely numeric screen names
+      "screen_name"
+    } else if (all(grepl("^[0-9]+$", x))) {
+      "user_id"
```

Comment:
Shouldn't we check the size too with "^[0-9]{64}$" ? This would eliminate some false results as `user_type("123")` being user_id or `user_type("1244564654456546465464646412445646544565464654646464124456465445654646546464641244564654456546465464646412445646544565464654646464")`. May be worth also to leave a comment with a link from where this magical 64 comes from.

Reply:
I don't think it's worth putting too much magic in here, although it might be better to construct this test using `bit64::as.integer64()` since that's what we really care about.

Reply:
Adding length check will reduce the potential errors for many users (specially those dealing with bots or newly created users which didn't set a user name), if you prefer use bit64 to check this.

Reply:
Maybe we should annotate rtweet results so that that screen names are always marked as screen names and this becomes less of a problem?

Reply:
Not sure how easy this would be, and if the user will see any benefit but I think it might be harder to implement, while checking this here the benefits are clearer and the change minimal.

Reply:
It's a bad idea to sometimes declare that a character vector is a vector of screen names and sometimes its ids based on some relatively subtle feature of the string — that makes the overall behaviour of the system harder to predict and to debug when things go wrong. 

If you think it's likely that users will have vector of screen names containing only numbers it is better to solve the problem upstream, where the vector is created.

Reply:
(Using `bit64::as.integer64()` doesn't work because it seems to use a very permissive parser, so I think it's best to leave this code as is)